### DISCUSSION TITLE
How do I write to file a DynamicDiscretisedDensity?

### DISCUSSION THREAD

**Discussion by AnderBiguri** [UCL/STIR#816] [Q&A] [answered]
I have 

     DynamicDiscretisedDensity dyn_gradient=this->_dyn_image_template;

and for debugging purposes I want to write to file. 

neither

     write_to_file("dyn_gradient0.hv",dyn_gradient[this->_patlak_plot_sptr->get_starting_frame()]);
nor

     write_to_file("dyn_gradient0.hv",dyn_gradient);

work, because: 

` error: there are no arguments to ‘write_to_file’ that depend on a template parameter, so a declaration of ‘write_to_file’ must be available`.

So how can I save this image to a file? Is there a general way that works for any image?

**Comment by AnderBiguri** [comment, marked as answer]
oops

`#include "stir/IO/write_to_file.h"
`

(I had it around a `#ifdef` that was not defined)

**Reply by robbietuk** [reply]
The first discussion, the question was asked and answered by the same person 😆
